src/utils/focusTrapComponent.ts

```diff
@@ -0,0 +1,72 @@
+import { FocusTrap as _FocusTrap, Options as FocusTrapOptions, createFocusTrap } from "focus-trap";
+import { FocusableElement, focusElement } from "./dom";
+
+/**
+ * Defines interface for components with a focus trap.
+ */
+export interface FocusTrapComponent {
+  /**
+   * The focus trap instance.
+   */
+  focusTrap: FocusTrap;
+
+  /**
+   * The focus trap element.
+   */
+  focusTrapEl: HTMLElement;
+}
+
+export type FocusTrap = _FocusTrap;
+
+/**
+ * Helper to set up focus trap component.
+ *
+ * @param component
+ */
+export function connectFocusTrap(component: FocusTrapComponent): void {
+  const { focusTrapEl } = component;
+
+  if (!focusTrapEl) {
+    return;
+  }
+
+  if (focusTrapEl.tabIndex == null) {
+    focusTrapEl.tabIndex = -1;
+  }
+
+  const focusTrapOptions: FocusTrapOptions = {
+    allowOutsideClick: true,
+    clickOutsideDeactivates: false,
+    delayInitialFocus: true,
+    escapeDeactivates: false,
+    returnFocusOnDeactivate: true,
+    fallbackFocus: focusTrapEl,
+    setReturnFocus: (el) => {
+      focusElement(el as FocusableElement);
+      return false;
+    },
+    tabbableOptions: {
+      getShadowRoot: true
+    }
+  };
+
+  component.focusTrap = createFocusTrap(focusTrapEl, focusTrapOptions);
```

Comment:
Could you add a spec test for these helpers?

Reply:
done